app/controllers/cases/letters_controller.rb

```diff
@@ -0,0 +1,30 @@
+class Cases::LettersController < ApplicationController
+  before_action :set_case
+
+  def new
+    @type = type_name(params[:type])
+    @letter = Letter.new(params[:letter_template_id])
+    @letter_templates = LetterTemplate.where(template_type: @type)
+  end
+
+  def render_letter
+    @type = params[:type]
+    unless params[:letter_form] and params[:letter_form][:letter_template_id]
+      flash[:notice] = 'Please select a template.'
+      redirect_to new_case_letters_path(@case.id, @type) and return
+    end
+    letter_template_id = params[:letter_form][:letter_template_id]
+    @letter = Letter.new(letter_template_id)
+    @letter_template = LetterTemplate.find(letter_template_id)
+  end
```

Comment:
Would prefer all this logic to either be in the `letter` model or in an appropriate service.

Don't use the `and` operator, use `&&` instead, unless you really do mean `and` ?

Should use strong parameters for consistency with codebase

Reply:
> Would prefer all this logic to either be in the letter model or in an appropriate service.
Will take a look at that, thanks...

> Should use strong parameters for consistency with codebase
This is what seemed strange to me - I went to add params but there are actually lots of places we just access the params directly. So this is IMO consistent with the codebase, which uses both approaches. I've ensured the params are not reflected back to the user in the page, and they're all used safely IMO. But I'll have another look at it, as I agree it's best-practice.

Reply:
Refactored the controller - I agree about possibly using a service for this, will take a look at that in the next PR when we add Word document export